### PR TITLE
Implement analyze methods send transaction accounts sign transaction send private transaction

### PR DESCRIPTION
**Pull request by novosandara**
# Description

Please provide a detailed description of what was done in this PR

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

# Documentation update

# Additional comments

### CODE REVIEW

File: consensus/polybft/wallet/account.go [hunk 1]
```diff
@@ -110,7 +110,12 @@ func (a *Account) Save(secretsManager secrets.SecretsManager) (err error) {
 }
 
 func (a *Account) GetEcdsaPrivateKey() (*ecdsa.PrivateKey, error) {
-	ecdsaRaw, err := a.Ecdsa.MarshallPrivateKey()
+	return AdaptECDSAPrivKey(a.Ecdsa)
+}
+
+// AdaptECDSAPrivKey converts ecdsa private key from wallet.Key to ecdsa.PrivateKey instance
```
Comment (Stefan-Ethernal) [inline consensus/polybft/wallet/account.go]: I've forgotten to update the comment, let's change it now.

```suggestion
// AdaptECDSAPrivKey converts ecdsa private key from crypto.ECDSAKey to ecdsa.PrivateKey instance
```

File: jsonrpc/eth_endpoint.go [hunk 2]
```diff
@@ -9,18 +9,28 @@ import (
...
+var (
+	errMissingPrivateKey = errors.New("local private key is undefined")
+)
+
 type ethTxPoolStore interface {
 	// AddTx adds a new transaction to the tx pool
 	AddTx(tx *types.Transaction) error
 
+	// CheckTx ensures the transaction conforms
+	CheckTx(tx *types.Transaction) error
```
Comment (Stefan-Ethernal) [inline jsonrpc/eth_endpoint.go]: This function is not being used anywhere, from what I can see. Delete it Mr. Batkovich?

```suggestion
```

File: jsonrpc/eth_txpool_test.go [hunk 3]
```diff
@@ -58,10 +105,18 @@ func (m *mockStoreTxn) AddTx(tx *types.Transaction) error {
 	return nil
 }
 
+func (m *mockStoreTxn) CheckTx(tx *types.Transaction) error {
+	return nil
+}
```
Comment (Stefan-Ethernal) [inline jsonrpc/eth_txpool_test.go]: Remove it?

```suggestion
```

File: txpool/txpool.go [hunk 4]
```diff
@@ -301,6 +301,16 @@ func (p *TxPool) SetSealing(sealing bool) {
...
 
+// CheckTx ensures the transaction conforms to specific
+// constraints before entering the pool.
+func (p *TxPool) CheckTx(tx *types.Transaction) error {
+	if err := p.validateTx(tx); err != nil {
+		return err
+	}
+
+	return nil
+}
```
Comment (Stefan-Ethernal) [inline txpool/txpool.go]: Remove it

```suggestion
```

File: jsonrpc/types.go [hunk 5]
```diff
@@ -137,7 +138,7 @@ type header struct {
 	BaseFee         argUint64   `json:"baseFeePerGas,omitempty"`
 }
 
-type accessListResult struct {
+type AccessListResult struct {
```
Comment (Stefan-Ethernal) [inline jsonrpc/types.go:141]: Why do you need it exported?

File: jsonrpc/mocks_test.go [hunk 6]
```diff
@@ -220,3 +224,17 @@ func (m *mockStore) GetStateSyncProof(stateSyncID uint64) (types.Proof, error) {
 func (m *mockStore) FilterExtra(extra []byte) ([]byte, error) {
 	return extra, nil
 }
+
+func setupSecretsManagerWithKey(t require.TestingT) *secrets.SecretsManagerMock {
+	ecdsaKey, err := wallet.GenerateKey()
```
Comment (Stefan-Ethernal) [inline jsonrpc/mocks_test.go:232]: Make this a test helper function
```suggestion
    t.Helper()
	ecdsaKey, err := wallet.GenerateKey()
```

File: jsonrpc/eth_endpoint.go [hunk 7]
```diff
@@ -1018,3 +1133,38 @@ func (e *Eth) FeeHistory(blockCount argUint64, newestBlock BlockNumber,
...
+		return nil, err
+	}
+
+	signedTx, err := e.txSigner.SignTx(tx, cryptoECDSAPrivKey)
+	if err != nil {
+		return nil, err
+	}
+
+	data := signedTx.MarshalRLP()
+	data[64] += 27 // Transform V from 0/1 to 27/28 according to the yellow paper
```
Comment (dusannosovic-ethernal) [inline jsonrpc/eth_endpoint.go]: I think this is not correct. It's correct when we have a byte array of the signature, where the 65th byte is V. However, when marshalling the entire transaction, I'm not sure that the 65th byte is associated with V.
@Stefan-Ethernal correct me if i am wrong.

Comment (novosandara) [reply jsonrpc/eth_endpoint.go]: Looks like you're right

**Pull request closed without merging**